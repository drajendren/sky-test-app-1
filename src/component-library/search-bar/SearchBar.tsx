import React from "react";
import { ResultContainer } from "..";
import SearchResponse from "../../interfaces/SearchResponse";
import styles from "./search-bar.module.scss";
export interface SearchBarProps {
  searchSuggestion: SearchResponse;
  searchValue: string;
  setSearchValue: any;
  handleClick: (e: React.SyntheticEvent) => void;
}

export const SearchBar = (props: SearchBarProps) => {
  const { searchValue, setSearchValue, handleClick, searchSuggestion } = props;
  const onHandleChange = (e: any) => {
    setSearchValue(e.target.value);
  };
  const handleEnter = (event: any) => {
    if (event.key === "Enter") {
      handleClick(event);
    }
  };

  return (
    <div>
      <div className={styles.searchBarContainer}>
        <div className={styles.autoComplete}>
          <input
            autoFocus
            className={styles.searchInputBox}
            type="text"
            value={searchValue}
            onChange={onHandleChange}
            onKeyDown={handleEnter}
          />
        </div>
        <button
          data-testid="searchButton"
          className={styles.searchButton}
          onClick={handleClick}
        >
          Search
        </button>
      </div>
      {searchSuggestion.results ? (
        <>
          <h4 data-testid="suggestedSearch">Suggested searches</h4>
          <div className={styles.suggestionsContainer}>
            {searchSuggestion?.results?.map((searchSuggestion, index) => {
              return (
                <div
                  data-testid={`search_suggestion_${index}`}
                  key={index}
                  className={styles.resultRow}
                >
                  <ResultContainer
                    className={"searchSuggestions"}
                    resultsObject={searchSuggestion}
                  />
                </div>
              );
            })}
          </div>
        </>
      ) : null}
    </div>
  );
};

export default SearchBar;
