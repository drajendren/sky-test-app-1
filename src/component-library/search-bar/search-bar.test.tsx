import React from "react";
import { fireEvent, render } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import {
  searchWithLessThanFiveCharactersData,
  searchWithResultsData,
} from "../../test-data/test-data";
import SearchBar from "./SearchBar";
import { BrowserRouter } from "react-router-dom";
import axios, { AxiosStatic } from "axios";
interface AxiosMock extends AxiosStatic {
  mockResolvedValue: Function;
  mockRejectedValue: Function;
}
jest.mock("axios");
const mockAxios = axios as AxiosMock;
describe("<SearchBar />", () => {
  test("It should render correctly", () => {
    const { baseElement } = render(
      <BrowserRouter>
        <SearchBar {...searchWithResultsData} />
      </BrowserRouter>
    );
    expect(baseElement).toMatchSnapshot();
  });
  test("When there are less than 5 characters in search, should be no suggestions displayed", () => {
    const { queryByTestId } = render(
      <BrowserRouter>
        <SearchBar {...searchWithLessThanFiveCharactersData} />
      </BrowserRouter>
    );
    const suggestedSearchResult = queryByTestId("search_suggestion_0");

    expect(suggestedSearchResult).toBeNull();
  });
  test("When there are more than 5 characters in search, should be no suggestions displayed", () => {
    const { getByTestId } = render(
      <BrowserRouter>
        <SearchBar {...searchWithResultsData} />
      </BrowserRouter>
    );
    const suggestedSearchResult = getByTestId("search_suggestion_0");
    expect(suggestedSearchResult).toBeTruthy();
  });
});
