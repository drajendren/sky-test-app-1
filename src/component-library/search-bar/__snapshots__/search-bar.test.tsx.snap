// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchBar /> It should render correctly 1`] = `
<body>
  <div>
    <div>
      <div
        class="searchBarContainer"
      >
        <div
          class="autoComplete"
        >
          <input
            class="searchInputBox"
            type="text"
            value="harry potter and the goble"
          />
        </div>
        <button
          class="searchButton"
          data-testid="searchButton"
        >
          Search
        </button>
      </div>
      <h4
        data-testid="suggestedSearch"
      >
        Suggested searches
      </h4>
      <div
        class="suggestionsContainer"
      >
        <div
          class="resultRow"
          data-testid="search_suggestion_0"
        >
          <a
            class="link"
            href="/674"
          >
            <button
              class="searchSuggestions"
            >
              Harry Potter and the Goblet of Fire
            </button>
          </a>
        </div>
      </div>
    </div>
  </div>
</body>
`;
